pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -457,19 +469,27 @@ func (d *DependenciesDistributor) isOrphanAttachedBindings(
 			if selector, err = metav1.LabelSelectorAsSelector(dependency.LabelSelector); err != nil {
 				return false, err
 			}
-			rawObject, err := helper.FetchResourceTemplate(d.DynamicClient, d.InformerManager, d.RESTMapper, workv1alpha2.ObjectReference{
-				APIVersion: resource.APIVersion,
-				Kind:       resource.Kind,
-				Namespace:  resource.Namespace,
-				Name:       resource.Name,
-			})
-			if err != nil {
-				// do nothing if resource template not exist.
-				if apierrors.IsNotFound(err) {
-					continue
+
+			if rawObject == nil {
+				rawObject, err = helper.FetchResourceTemplate(d.DynamicClient, d.InformerManager, d.RESTMapper, workv1alpha2.ObjectReference{
+					APIVersion: resource.APIVersion,
+					Kind:       resource.Kind,
+					Namespace:  resource.Namespace,
+					Name:       resource.Name,
+				})
+				if err != nil {
+					// return nil err if resource template not exist.
+					if apierrors.IsNotFound(err) {
+						return false, nil
+					}
+					return false, err
+				}
+				// return nil err if resource template is being deleted.
+				if !rawObject.GetDeletionTimestamp().IsZero() {
+					return false, nil
```

Comment:
I understand that placing this part of the code outside the for loop (before the loop starts) would be easier to understand.